From 99eaf1337989445be6d86758a349d1fe84d0a7a1 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Sun, 3 Apr 2016 14:37:26 +0300
Subject: [PATCH 080/239] axi: Set all IOs AXI transactions to non-secure

Currently, the system is configured to work in non-secure mode,
thus IO DMA transactions must be configured to work in non-secure as well.

Change-Id: I8f7f5622790e517b9255a81d0147d2a9633cb35a
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28736
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 plat/marvell/a8k/a70x0/cp110_setup.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plat/marvell/a8k/a70x0/cp110_setup.c b/plat/marvell/a8k/a70x0/cp110_setup.c
index e066c29..23747db 100644
--- a/plat/marvell/a8k/a70x0/cp110_setup.c
+++ b/plat/marvell/a8k/a70x0/cp110_setup.c
@@ -52,6 +52,12 @@
 #define MVEBU_AXI_ATTR_AWDOMAIN_OFFSET		28
 #define MVEBU_AXI_ATTR_AWDOMAIN_MASK		(0x3 << MVEBU_AXI_ATTR_AWDOMAIN_OFFSET)
 
+/* AXI Protection bits */
+#define MVEBU_AXI_PROT_BASE			(MVEBU_CP0_REGS_BASE + 0x441200)
+#define MVEBU_AXI_PROT_REG(index)		((index <= 6) ? (MVEBU_AXI_PROT_BASE + 0x4 * index) : \
+							(MVEBU_AXI_PROT_BASE + 0x20 + 0x4 * (index - 7)))
+#define MVEBU_AXI_PROT_REGS_NUM			(9)
+
 /* AXI to MBUS bridge registers */
 #define MVEBU_AMB_IP_BRIDGE_WIN_REG(win)	(MVEBU_AMB_IP_BASE + (win * 0x8))
 #define MVEBU_AMB_IP_BRIDGE_WIN_EN_OFFSET	0
@@ -109,6 +115,10 @@ void cp110_axi_attr_init(void)
 		}
 	}
 
+	/* Set all IO's AXI attribute to non-secure access. */
+	for (index = 0; index < MVEBU_AXI_PROT_REGS_NUM; index++)
+		mmio_write_32(MVEBU_AXI_PROT_REG(index), 0x3);
+
 	return;
 }
 
-- 
1.9.1

